### PR TITLE
pacific: nofail option in fstab not supported 

### PR DESCRIPTION
**Pull request by leonid-s-usov**
Backport tracker: https://tracker.ceph.com/issues/62427
Parent tracker ticket: https://tracker.ceph.com/issues/58394
Parent PR: https://github.com/ceph/ceph/pull/52834

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by leonid-s-usov** [comment]
jenkins retest this please

**Comment by leonid-s-usov** [comment]
jenkins retest this please

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by leonid-s-usov** [comment]
jenkins test windows

**Comment by leonid-s-usov** [comment]
jenkins test dashboard cephadm

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by leonid-s-usov** [comment]
@vshankar I decided to rebase this over the latest pacific to see if that helps pass the checks

File: src/mount.fuse.ceph [hunk 1]
```diff
@@ -37,8 +37,11 @@ def ceph_options_compat(device):
     return [ 'ceph.' + opt for opt in device.split(',') ]
 
 def fs_options(opts, ceph_opts):
-    # strip out noauto and _netdev options; libfuse doesn't like it
-    strip_opts = ['defaults', 'noauto', '_netdev']
+    # - strip out noauto and _netdev options; libfuse doesn't like it
+    # - nofail option is also not recognized by libfuse.
+    #   Starting with fuse 3.2.2 the option is also ignored by mount.fuse, see
+    #   https://github.com/libfuse/libfuse/commit/a83cd72f641671b71b8268b1765e449cae071f3e
```
Comment (mchangir) [inline src/mount.fuse.ceph:43]: `Starting with fuse 3.2.2 the option is also ignored by mount.fuse ...`

which option is also ignored by `mount.fuse` ?

Comment (leonid-s-usov) [reply src/mount.fuse.ceph:43]:
```
    # - strip out noauto and _netdev options; libfuse doesn't like it
    # - nofail option is also not recognized by libfuse.
    #   Starting with fuse 3.2.2 the option is also ignored by mount.fuse, see
    #   https://github.com/libfuse/libfuse/commit/a83cd72f641671b71b8268b1765e449cae071f3
```
I've added `-` to separate the list items, but that wasn't helpful, apparently :). "the option" refers to `nofail` from the line above.

**Comment by rishabh-d-dave** [comment]
make check job failed but logs are absent (https://jenkins.ceph.com/job/ceph-pull-requests/123270/). Re-running again.

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
Links to parent tracker ticket and parent PR was missing, I've added it

**Review by rishabh-d-dave — COMMENTED** [review]
@yuriw @vshankar 
FS approved. QA run was successful - https://tracker.ceph.com/projects/cephfs/wiki/Pacific#8-Nov-2023.

**Comment by yuriw** [comment]
@leonid-s-usov this is ready for merge when all comments resolved
ref: https://trello.com/c/kcV6kqd4

**Comment by vshankar** [comment]
@rishabh-d-dave please approve and merge if nothing is pending.

**Review by rishabh-d-dave — APPROVED** [review]
Approving on behalf of CephFS

**Pull request merged**